### DISCUSSION TITLE
How to emit from one connection to another

### DISCUSSION THREAD

**Discussion by datonefaridze** [miguelgrinberg/python-socketio#847] [Q&A]
Hello, I have 2 connections (kind like a middlware), i want to receive data from one connection and send it to another, it seems that code fails in this case. my code:
```
@sio_featurize.event
def streamFromPy(data):
    print("streamFromPy")
    sio.emit("test", {'data': [1,2,3]})) 
```
and sio is one connection, and sio_featuze is another connection.

**Comment by miguelgrinberg** [comment]
What do you mean by "connection" here? A client or a server?

Assuming these are clients, you cannot emit from one client to another, that is not how the Socket.IO protocol works. Clients can only emit to the server, so the way you do this is by having client A send the data to the server, which then in turn relays it to client B.

**Reply by datonefaridze** [reply]
Thansk for the quick answer. sio is socketio.Server and sio_featurize is  socketio.Client(). i am waiting for sio_featurize to recieve data and then i want to send it via sio  connection

**Reply by datonefaridze** [reply]
also i am recieving data from python script and sending it to react script (i don't know if it matters or not)

**Reply by datonefaridze** [reply]
@miguelgrinberg

**Reply by miguelgrinberg** [reply]
I'm sorry but this is still unclear to me. As I said before, your client can only emit to a server it is connected to. The server can only emit to one or more connected clients. That is pretty much it, you have to work with that to build the flow of data that you need.

**Reply by datonefaridze** [reply]
![image](https://user-images.githubusercontent.com/52849166/148943725-60fd42f5-acf9-45b4-83c8-87d739774369.png)
this is my flow. basically i want to know how can i emit data from @sio_featurize.event using sio.emit("test", {'data': [1,2,3]}) ? 
my code looks like this:
```
sio = socketio.Server(cors_allowed_origins='*')
sio_featurize = socketio.Client()
sio_featurize.connect('http://127.0.0.1:8080') #172.17.0.2:8080 localhost

app = socketio.WSGIApp(sio)


@sio.event
def streamToNode(sid, data):
    print("streamToNodeFile")

    sio_featurize.emit("streamToPy", data)

@sio_featurize.event
def streamFromPy(sid, data):
    sio.emit("test", {'data': [1,2,3]})
```

**Reply by datonefaridze** [reply]
please help me, i have been trying to do it for about a week @miguelgrinberg

**Reply by datonefaridze** [reply]
on the picture there should be *sio

**Reply by miguelgrinberg** [reply]
You forgot to mention what's not working and need help with. The code that you have above appears to be doing the cross-connection communication.

**Reply by datonefaridze** [reply]
when i emit sio.emit("test", {'data': [1,2,3]}), it doesn't go to frontend server. frontend server connection is extremelysimple:
```
    socketClient.on("test", (data) => {
      console.log("socketClient");
    });
```
   strange thing is when i emit data from 
```
   @sio.event
def streamToNode(sid, data):
    print("streamToNodeFile")
    sio.emit("test", {'data': [1,2,3]})
    sio_featurize.emit("streamToPy", data)
```
it goes to frontend server (sio.emit is in streamToNode function instead of streamFromPy)

**Reply by miguelgrinberg** [reply]
There is really nothing in one function vs the other that can change how the emit works, so this has to be something else. I noticed that you don't have any print statements on the function that does not emit. Maybe the function isn't called? I think you need to debug this putting prints everywhere to figure it out. If it helps, you can also enable logging for the Socket.IO functions. See the troubleshooting section of the docs for instructions.
